### PR TITLE
Swagger update for Azure Active Directory Integration for Azure Database for Postgres and MySQL

### PR DESCRIPTION
**Pull request by shinilm**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@shinilm can you please sign the CLA?

**Comment by jhendrixMSFT** [comment]
Also looks like two new ARM errors have been introduced which will need to be fixed.
```
| R2062 - XmsResourceInPutResponse |  
| The 200 response model for an ARM PUT operation must have x-ms-azure-resource extension set to true in its hierarchy. Operation: 'ServerAdministrators_CreateOrUpdate' Model: 'ServerAdministratorResource'. |  
| at specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json:1108 |  
|   |  
| R2020 - RequiredPropertiesMissingInResourceModel |  
| Model definition 'ServerAdministratorResource' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly. |  
| at specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2017-12-01/postgresql.json:2422
```

**Comment by shinilm** [comment]
@jhendrixMSFT,  Thanks for the review and I have addressed the same

**Comment by jhendrixMSFT** [comment]
Can you please also fix the model validation failures listed [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=73079&view=logs&j=8d2f9c49-cd83-5a9d-f3fe-2fd651afa742).  Looks like the example files need a small fix.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/examples/VirtualNetworkRulesDelete.json [hunk 1]
```diff
@@ -10,11 +10,7 @@
     "200": {
       "body": ""
```
Comment (jhendrixMSFT) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/examples/VirtualNetworkRulesDelete.json]: Model validation is also complaining about 200 (comment applies to all three examples).

Comment (shinilm) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/examples/VirtualNetworkRulesDelete.json]: This has been resolved

**Comment by jhendrixMSFT** [comment]
Thanks.  This looks good from SDK perspective, waiting for ARM to sign off.

**Comment by jhendrixMSFT** [comment]
@sanjaiganesh could somebody from ARM please review?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json [hunk 2]
```diff
@@ -1072,6 +1072,178 @@
...
+              "$ref": "#/definitions/ServerAdministratorResource"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/ServerAdministratorResource"
+            }
+          },
+          "202": {
```
Comment (KrisBash) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: When would a 202 be used vs 201?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json [hunk 3]
```diff
@@ -1072,6 +1072,178 @@
...
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          }
+        ],
+        "responses": {
```
Comment (KrisBash) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: For all responses, please add a "default" response for error cases. This should implement the standard error contract. You can link the definition here: https://github.com/Azure/azure-rest-api-specs/blob/df118006164af427f4d441c20e9a570c28505876/specification/common-types/resource-management/v1/types.json#L272

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/examples/ServerAdminGet.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  "parameters": {
+    "subscriptionId": "ffffffff-ffff-ffff-ffff-ffffffffffff",
+    "resourceGroupName": "testrg",
+    "serverName": "mysqltestsvc4",
+    "administratorName": "activeDirectory",
+    "api-version": "2018-06-01-privatepreview"
+  },
+  "responses": {
+    "200": {
+      "body": {
```
Comment (KrisBash) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/examples/ServerAdminGet.json]: Example responses are missing "type" property

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json [hunk 5]
```diff
@@ -2272,6 +2444,68 @@
...
       },
       "description": "An error response from the Batch service."
+    },
+    "ServerAdministratorProperties": {
+      "properties": {
+        "administratorType": {
+          "type": "string",
+          "description": "The type of administrator.",
+          "enum": [
+            "ActiveDirectory"
```
Comment (KrisBash) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: This enum implies that there might be other types in the future, but the schema is not polymorphic. If you plan to support future types, you may want to use a 'discriminator' and use a unique definition object bound to the enum value.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/mysql.json [hunk 6]
```diff
@@ -1072,6 +1072,178 @@
...
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          }
+        ],
+        "responses": {
```
Comment (KrisBash) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/mysql.json]: Same as above concerning "default" response in each api

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/mysql.json [hunk 7]
```diff
@@ -1072,6 +1072,178 @@
...
+              "$ref": "#/definitions/ServerAdministratorResource"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/ServerAdministratorResource"
+            }
+          },
+          "202": {
```
Comment (KrisBash) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/mysql.json]: Do you return both 201 or 202?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/mysql.json [hunk 8]
```diff
@@ -2832,6 +3004,68 @@
         }
       },
       "description": "An error response from the Batch service."
+    },
+     "ServerAdministratorProperties": {
+      "properties": {
+        "administratorType": {
```
Comment (KrisBash) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/mysql.json]: Same as above concerning consideration of a polymorphic schema

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/examples/ServerAdminCreateUpdate.json [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        "login": "bob@contoso.com",
+        "sid": "c6b82b90-a647-49cb-8a62-0d2d3cb7ac7c",
+        "tenantId": "c6b82b90-a647-49cb-8a62-0d2d3cb7ac7c"
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/ffffffff-ffff-ffff-ffff-ffffffffffff/resourceGroups/testrg/providers/Microsoft.DBforMySQL/servers/mysqltestsvc4/administrators/activeDirectory",
```
Comment (KrisBash) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/examples/ServerAdminCreateUpdate.json]: Missing "type"

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comments

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json (hunk 5 above)
Comment (shinilm) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: Discussed with the team and there will not be any other type in future.  So this is a won't fix

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/mysql.json (hunk 8 above)
Comment (shinilm) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2018-06-01-privatepreview/mysql.json]: Discussed with the team and there will not be any other type in future.  So this is a won't fix

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json (hunk 2 above)
Comment (shinilm) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2017-12-01-preview/mysql.json]: Removed the 201's

**Comment by yungezz** [comment]
hi @shinilm is this PR ready to merge?

**Comment by shinilm** [comment]
Yes. This is ready for merge

Thanks
Shinil

On Sep 26, 2019, at 12:44 AM, Yunge Zhu <notifications@github.com> wrote:

﻿

hi @shinilm<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fshinilm&data=02%7C01%7Cshinim%40microsoft.com%7Cd33b8848213f47f8366b08d74255639f%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637050806817900135&sdata=7A3S08e55Fm6XlkN%2BkiZyE3Xv2C9ZlatbpwjSZ7l43k%3D&reserved=0> is this PR ready to merge?

**Comment by yungezz** [comment]
approve since ARM signed off, and do merge.

**Review by yungezz — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
@shinilm is this functionality live?

**Comment by jhendrixMSFT** [comment]
@yungezz can yo confirm the functionality is live?

**Comment by shinilm** [comment]
> hi @shinilm is this PR ready to merge?



> hi @shinilm is this PR ready to merge?

Yes this is live

**Comment by shinilm** [comment]
> hi @shinilm is this PR ready to merge?

Yes this is ready to merge

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**